02-Generics/Generics/Generics.cs

```diff
@@ -91,9 +98,72 @@ public static class ArrayExtentions {
 		///     { 1, "a", false },
 		///   }
 		/// </example>
-		public static void SortTupleArray<T1, T2, T3>(this Tuple<T1, T2, T3>[] array, int sortedColumn, bool ascending) {
+		public static void SortTupleArray<T1, T2, T3>(this Tuple<T1, T2, T3>[] array, int sortedColumn, bool ascending) 
+				where T1: IComparable
+				where T2:IComparable
+				where T3:IComparable
+		{
 			// TODO :SortTupleArray<T1, T2, T3>
 			// HINT : Add required constraints to generic types
+			bool swap = false;
+
+			if (sortedColumn >= 0 && sortedColumn <=2)
+			{
+				for (int i = 0; i < array.Length; i++)
+					for (int j = 0; j < array.Length; j++)
+					{
+						switch (sortedColumn)
+						{
+							case 0:
+								switch (ascending)
+								{
+									case true:
+										if (array[i].Item1.CompareTo(array[j].Item1) < 0)
+											swap = true;
+										break;
+									case false:
+										if (array[i].Item1.CompareTo(array[j].Item1) > 0)
```

Comment:
instead of adding case true or false it can be done in one line in the end after sorting